petastorm/spark/spark_dataset_converter.py

```diff
@@ -517,6 +536,19 @@ def wait_for_file(path):
         pool.join()
 
 
+def _check_dataset_file_median_size(url_list):
+    fs, path_list = get_filesystem_and_path_or_paths(url_list)
+    # TODO: also check other file system.
+    if isinstance(fs, LocalFileSystem):
```

Comment:
Would it make sense to check also for non-local file systems? Not sure how fast does it work with fuse - would it make sense to acquire file sizes on a threadpool?

Reply:
* check also for non-local file systems make sense, but pyarrow filesystem interface do not implement the `stat` method. It need additional work so we could do it later.
* acquire file sizes by a threadpool make sense.